webapp/channels/src/actions/views/root.ts

```diff
@@ -20,22 +34,56 @@ const pluginTranslationSources: Record<string, TranslationPluginFunction> = {};
 
 export type TranslationPluginFunction = (locale: string) => Translations
 
-export function loadConfigAndMe(): ThunkActionFunc<Promise<{config?: ClientConfig; isMeLoaded: boolean}>> {
-    return async (dispatch) => {
-        const results = await Promise.all([
-            dispatch(getClientConfig()),
-            dispatch(getLicenseConfig()),
-        ]);
-
-        let isMeLoaded = false;
-        if (document.cookie.includes('MMUSERID=')) {
-            const dataFromLoadMe = await dispatch(loadMe());
-            isMeLoaded = dataFromLoadMe?.data ?? false;
+/**
+ * This function meant to be used in root.tsx component loads config, license and if user is logged in, it loads user and its related data.
+ */
+export function loadConfigAndMe(): ThunkActionFunc<Promise<{isLoaded: boolean; isMeRequested?: boolean}>> {
+    return async (dispatch, getState) => {
+        // attempt to load config and license regardless if user is logged in or not
+        try {
+            await Promise.all([
+                dispatch(getClientConfig()),
+                dispatch(getLicenseConfig()),
+            ]);
+        } catch (error) {
+            dispatch(logError(error as ServerError));
+            return {
+                isLoaded: false,
+            };
+        }
+
+        // Return early if user is not logged in
+        if (!doesCookieContainsMMUserId()) {
+            return {
+                isLoaded: true,
+                isMeRequested: false,
+            };
+        }
+
+        // Load user and its related data now that we know that user is logged in
+        const serverVersion = getState().entities.general.serverVersion || Client4.getServerVersion();
+        dispatch({type: GeneralTypes.RECEIVED_SERVER_VERSION, data: serverVersion});
+
+        try {
+            await Promise.all([
+                dispatch(getMe()),
+                dispatch(getMyPreferences()),
+                dispatch(getMyTeams()),
+                dispatch(getMyTeamMembers()),
```

Comment:
To avoid duplicating most of `loadMe` here, could we just replace `loadMe` with this method everywhere that it's used? Alternatively, we could pass a `skipLoadingConfig` parameter into `loadMe` an have this just dispatch `loadMe`, but I don't think that's worth the extra complexity

Reply:
I thought about this a lot but finally decided to create just another function for this as you mentioned its worth the extra complexity just for usage in one plance